packages/apollo-gateway/src/index.ts

```diff
@@ -32,26 +33,47 @@ export interface GraphQLService {
 
 export type ServiceEndpointDefinition = Pick<ServiceDefinition, 'name' | 'url'>;
 
-export interface GatewayConfigBase {
+interface GatewayConfigBase {
   debug?: boolean;
   // TODO: expose the query plan in a more flexible JSON format in the future
   // and remove this config option in favor of `exposeQueryPlan`. Playground
   // should cutover to use the new option when it's built.
   __exposeQueryPlanExperimental?: boolean;
   buildService?: (definition: ServiceEndpointDefinition) => GraphQLDataSource;
-  serviceList?: ServiceEndpointDefinition[];
 }
 
+export interface ConcreteGatewayConfig extends GatewayConfigBase {
+  serviceList: ServiceEndpointDefinition[];
+}
+
+export interface HostedGatewayConfig extends GatewayConfigBase {
+  apiKey?: string;
+  tag?: string;
+  federationVersion?: number;
```

Comment:
@JacksonKearl I know that @trevor-scheer has been thinking through fallbacks and control overrides for fetching from the managed system in case of failure. I wonder if we should merge all three of these configs together into a simpler apiKey + serviceList / way to fetch services?

Can you sync with him on those ideas?